apisix/consumer.lua

```diff
@@ -94,6 +97,26 @@ function _M.consumers()
 end
 
 
+local function create_consume_cache(consumers_conf, key_attr, consumer_names)
+    core.table.clear(consumer_names)
+
+    for _, consumer in ipairs(consumers_conf.nodes) do
+        core.log.info("consumer node: ", core.json.delay_encode(consumer))
+        consumer_names[consumer.auth_conf[key_attr]] = consumer
+    end
+
+    return consumer_names
+end
+
+
+function _M.consumers_kv(plugin_name, consumer_conf, key_attr, consumer_names)
```

Comment:
Why do we need to pass a consumer_names and fill it?

Reply:
Seems unnecessary, I'll use a temporary consumer_names table